xmppserver/src/main/java/org/jivesoftware/openfire/group/DefaultGroupProvider.java

```diff
@@ -204,7 +235,10 @@ public void deleteGroup(String groupName) {
             // Remove the group entry.
             pstmt = con.prepareStatement(DELETE_GROUP);
             pstmt.setString(1, groupName);
-            pstmt.executeUpdate();
+            int groupsDeleted = pstmt.executeUpdate();
+            if (groupsDeleted == 0){
+                throw new GroupNotFoundException("Did not find group to delete");
+            }
```

Comment:
Is this better than an explicit check for the existence of the group? I realised I've used a mix of both. This way is cheaper, but it's less explicit.

Reply:
My primary concern would be to not change the API if we don't have to. This kind of thing is very likely to be implemented in third party integrations. We shouldn't break those for the sake of making things more consistent. That said, I wish we would've had more consistency in the first place. In this specific example, there also is an argument to be made for silently ignoring the fact that a group that is being deleted doesn't exist in the first place: the end result is going to be the same. That's again that we really _should_ be doing (or _not_ be doing) consistently, but alas...

Reply:
I believe that since this feedback, we've opted to add these changes to a non-patch release, and live with the consequences.